gdk/commands/component/PublishCommand.py

```diff
@@ -0,0 +1,383 @@
+                    logging.info("Not creating an artifacts bucket as it already exists.")
+                    return
+                logging.error("Cannot create the artifacts bucket '{}' as it is already owned by you in other region.")
+            raise Exception(e)
+        except Exception as e:
+            logging.error("Failed to create the bucket '{}' in region '{}'".format(bucket, region))
+            raise Exception(e)
+        logging.info("Successfully created the artifacts bucket '{}' in region '{}'".format(bucket, region))
+
+    def bucket_exists_in_same_region(self, bucket, region):
+        """
+        Checks if region of the bucket is same as the given region.
+
+
+        Parameters
+        ----------
+            bucket(string): Name of the bucket to create.
+            region(string): Name of the region to check.
+
+        Returns
+        -------
+            bool: Returns true if the bucket region is same as the region in check. Else false.
+        """
+        try:
+            response = self.service_clients["s3_client"].get_bucket_location(Bucket=bucket)
+            if response["LocationConstraint"] == region:
+                return True
+            return False
+        except Exception as e:
+            raise Exception("Unable to fetch the location of the bucket '{}'.\n{}".format(bucket, e))
+
+    def create_gg_component(self, c_name, c_version):
+        """
+        Creates a GreengrassV2 private component using its recipe.
+
+        Raises an exception if the recipe is invalid or the request is not successful.
+
+        Parameters
+        ----------
+            c_name(string): Name of the component to create.
+            c_version(string): Version of the component to create.
+
+        Returns
+        -------
+            None
+        """
+        publish_recipe_file = self.project_config["publish_recipe_file"]
+        with open(publish_recipe_file) as f:
+            try:
+                self.service_clients["greengrass_client"].create_component_version(inlineRecipe=f.read())
+            except Exception as e:
+                logging.error("Failed to create the component using the recipe at '{}'.".format(publish_recipe_file))
+                raise Exception("Creating private version '{}' of the component '{}' failed.\n{}".format(c_version, c_name, e))
+            logging.info("Created private version '{}' of the component in the account.'{}'.".format(c_version, c_name))
+
+    def get_next_version(self):
+        """
+        Calculates the next patch version of the component if it already exists in the account. Otherwise, it uses 1.0.0 as
+        the fallback version.
+
+        Parameters
+        ----------
+            None
+
+        Returns
+        -------
+            version(string): Version of the component to create.
+        """
+        fallback_version = "1.0.0"
+        logging.debug("Fetching private components from the account.")
+        try:
```

Comment:
I've noticed that `NEXT_PATCH` doesn't work if an existing version isn't already in GCS. Can we set a baseline for that, e.g. to create 1.0.0?

Reply:
It should work. https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/main/gdk/commands/component/publish.py#L186-L191

Reply:
Hmm. Maybe I was using an older version.